src/Microsoft.Sbom.Api/Executors/IJsonSerializationStrategy.cs

```diff
@@ -2,31 +2,68 @@
+    /// <returns></returns>
     public bool AddToFilesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config);
 
+    /// <summary>
+    /// Adds the config to the list of configs that support packages.
+    /// </summary>
+    /// <param name="elementsSupportingConfigs"></param>
+    /// <param name="config"></param>
+    /// <returns></returns>
     public bool AddToPackagesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config);
 
+    /// <summary>
+    /// Adds the config to the list of configs that support relationships.
+    /// </summary>
+    /// <param name="elementsSupportingConfigs"></param>
+    /// <param name="config"></param>
+    /// <returns></returns>
     public bool AddToRelationshipsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config);
 
+    /// <summary>
+    /// Adds the config to the list of configs that support external document references.
+    /// </summary>
+    /// <param name="elementsSupportingConfigs"></param>
+    /// <param name="config"></param>
+    /// <returns></returns>
     public bool AddToExternalDocRefsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config);
 
-    public void AddHeadersToSbom(ISbomConfigProvider sbomConfigs, ISbomConfig config)
-    {
-    }
-
-    public Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
-        ISbomConfig sbomConfig,
-        string spdxManifestVersion,
-        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
-        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
-        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
-        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator);
+    /// <summary>
+    /// Adds a metadata dictionary as a JSON element to the SBOM.
+    /// This is only used for SPDX 2.2 SBOM generation, metadata is added differently for SPDX 3.0 and above.
+    /// </summary>
+    /// <param name="sbomConfigs"></param>
+    /// <param name="config"></param>
+    public void AddMetadataToSbom(ISbomConfigProvider sbomConfigs, ISbomConfig config);
+
+    /// <summary>
+    /// Starts an array with a graph header that is only used in SPDX 3.0 and above.
+    /// This does not apply to SPDX 2.2 SBOM generation.
+    /// </summary>
+    /// <param name="manifestInfosFromConfig"></param>
+    /// <param name="sbomConfigs"></param>
+    public void StartGraphArray(IList<ManifestInfo> manifestInfosFromConfig, ISbomConfigProvider sbomConfigs);
+
+    /// <summary>
+    /// Ends an array with a graph header that is only used in SPDX 3.0 and above.
+    /// This does not apply to SPDX 2.2 SBOM generation.
+    /// </summary>
+    /// <param name="manifestInfosFromConfig"></param>
+    /// <param name="sbomConfigs"></param>
+    public void EndGraphArray(IList<ManifestInfo> manifestInfosFromConfig, ISbomConfigProvider sbomConfigs);
```

Comment:
IEnumerable instead of IList (will need to propagate to implementations)
```suggestion
    public void EndGraphArray(IEnumerable<ManifestInfo> manifestInfosFromConfig, ISbomConfigProvider sbomConfigs);
```